src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -21,6 +21,18 @@
 MatchableType = Union[Matchable, Type[BaseSegment]]
 
 
+def fluff_type(self):
+    if hasattr(self, "_cls"):
+        return self._cls
+    return type(self)
+
+
+def fluff_issubclass(cls, class_or_tuple):
+    if hasattr(cls, "cls"):
+        cls = getattr(cls, "cls")
+    return issubclass(cls, class_or_tuple)
+
```

Comment:
We should tie these into `BaseSegment.is_type()` so that the functionality is consistent. Either these become `staticmethod` on `BaseSegment`, or these methods move to a `helpers.py` file and then `.is_type()` calls them.